pyof/foundation/basic_types.py

```diff
@@ -517,41 +506,18 @@ def pack(self, value=None):
 
         if value is None:
             value = self
-        else:
-            container = type(self)(items=None)
-            container.extend(value)
-            value = container
 
         bin_message = b''
         try:
             for item in value:
+                if not isinstance(item, self._pyof_class):
```

Comment:
same as previous comment:


1: This is the same behavior of any of the other classes. UBInt8('a') for example is perfectly valid, and it only complains when trying to pack. The list is just keeping the same behavior.

2: This allows FixedTypeList(UBInt8, [1,2,3]) for example. Checking the class does not.